docs/dev/architecture.md

```diff
@@ -93,7 +96,148 @@ The AWS Deadline Cloud Worker Agent has the following key responsibilities:
     -   Stream worker agent logs to CloudWatch Logs
     -   Capture session output and stream to CloudWatch logs
 
-## 2. Software Architecture
+## 2. Worker Host Architecture
+
+The worker host is the compute environment where the worker agent runs and executes jobs. This section details the key aspects of the worker host architecture.
+
+![High-Level Architecture](./images/worker_host_architecture.drawio.svg)  
+<small>*Diagram can be edited using draw.io*</small>
+
+The type of host machine is abstract. Examples of worker hosts include:
+
+*   EC2 instances
+*   physical on-premise servers
+*   virtual machines
+*   containers
+
+### 2.1. Worker Agent Installation
+
+The installation of the worker agent is a two step process. First, the Python package must be
+downloaded. An example for this using `pip`:
+
+```sh
+pip install dealdine-cloud-worker-agent
+```
+
+After the Python package is installed, the worker agent installer can be run to complete the setup.
+The Python package provides a `install-deadline-worker` command-line entrypoint for this purpose.
+To learn about the available arguments, run:
+
+```sh
+install-deadline-worker --help
+```
+
+The `install-deadline-worker` program:
+
+- Creates necessary directories and file structures required by the worker agent at run-time
+- Creates required operating system users and groups
+- Sets up required file-system permissions and ownership
+- Grants the required operating system user privileges
+- Sets up operating system services for automatic startup and crash recovery
+
+### 2.2. Operating System Users and Groups
+
+-   **Agent User**:
+    -   Dedicated system user that the worker agent process runs as
+    -   Defaults to `deadline-worker-agent` on Linux and `deadline-worker` on Windows
+    -   Limited privileges following the principle of least privilege
+    -   No login shell access for security
+
+-   **Job Users**:
+    -   Queues can specify a "job run as user" that jobs will run as.
+    -   The worker host admin can configure this user's permissions on the worker host as required.
+    -   Separate user contexts for jobs from different queues
```

Comment:
```suggestion
    -   Enables separate user contexts for jobs from different queues
```

Reply:
Honestly this one is really nitpicky so feel free to ignore

Reply:
Applied